docs/MSAL_2x_Migration_Guide.md

```diff
@@ -0,0 +1,568 @@
+| **Error Domain**   | `MSALErrorDomain`       |
+| **Error Code**     | `-50000`                |
+| **Internal Error Code** | `-42011`           |
+| **Description**    | Varies depending on the validation failure (e.g., missing scheme, mismatched bundle ID, invalid host) |
+
+#### Common Redirect URI Errors
+
+| Error Scenario               | Example                      | Error Description                                                                                                                                                        |
+|------------------------------|------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **Empty URI**                | `nil`                        | "The provided redirect URI is nil or empty. Please ensure the redirect URI follows the valid format: `msauth.<bundle_id>://auth.`"                                      |
+| **Missing scheme**           | `://auth` (missing scheme)   | "The provided redirect URI is missing a scheme. Please ensure the redirect URI follows the valid format: `msauth.<bundle_id>://auth`. Register your scheme in `Info.plist` under `CFBundleURLSchemes`." |
+| **HTTP/HTTPS scheme**        | `http://yourapp.com`         | "The provided redirect URI uses an unsupported scheme (`http(s)://host`). Please ensure the redirect URI follows the valid format: `msauth.<bundle_id>://auth.`"        |
+| **Bundle ID mismatch**       | `msauth.wrong.bundle.id://auth` | "The provided redirect URI uses MSAL format but the bundle ID does not match the app’s bundle ID. Please ensure the redirect URI follows the valid format: `msauth.<bundle_id>://auth.`" |
+| **Missing host**             | `msauth.appid:/`             | "The provided redirect URI is missing the required host component `auth`. Please ensure the redirect URI follows the valid format: `msauth.<bundle_id>://auth.`"      |
+| **Legacy OAuth2 out-of-band format** | `urn:ietf:wg:oauth:2.0:oob` | "The provided redirect URI `urn:ietf:wg:oauth:2.0:oob` is not supported. Please ensure the redirect URI follows the valid format: `msauth.<bundle_id>://auth.`"         |
+| **Unknown invalid format**   | Any other invalid URI        | "The provided redirect URI is invalid. Please ensure the redirect URI follows the valid format: `msauth.<bundle_id>://auth.`"                                           |
+
+### 2. Valid Parent View Controller (UI Anchor Window) Requirement for Interactive Token Requests
+
+#### What Changed
+
+Starting with **MSAL 2.x** for **iOS** and **macOS**, providing a valid parent view controller is **mandatory** for any interactive authentication flow.  
+In **MSAL 1.x**, it was optional for macOS.
+
+A valid parent view controller must be **non-nil** and its view must be attached to a valid **window** (i.e., `parentViewController.view.window != nil`).
```

Comment:
What happens when parentViewCotroller is defined but view.window is not defined? Is there an error thrown?

Reply:
Yes, mentioned errors below in section: #### Common Invalid Parent View Controller Errors